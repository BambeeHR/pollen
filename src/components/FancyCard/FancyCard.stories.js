import FancyCard from './FancyCard.vue';

export default {
  title: 'FancyCard',
  component: FancyCard,
  parameters: {
    componentSubtitle: 'Basic card component.',
  },
};

export const Default = () => {
  return {
    components: { FancyCard },
    template: `
      <FancyCard>
        <div class="max-w-xl m-auto text-center">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod 
          tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim 
          veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea 
          commodo consequat. Duis aute irure dolor in reprehenderit in voluptate 
          velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint 
          occaecat cupidatat non proident, sunt in culpa qui officia deserunt 
          mollit anim id est laborum.
        </div>
        <template #footer>
          This is a footer.
        </template>
      </FancyCard>
    `,
  };
};

/* eslint-disable no-param-reassign */
[Default].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['FancyCard.test.js'],
    },
  };
});
