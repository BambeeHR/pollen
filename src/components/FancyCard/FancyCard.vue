<template>
  <BaseCard
    class="fancy-card"
    :class="{ 'fancy-card--ornate': ornate }"
    v-bind="$attrs"
  >
    <template #header>
      <div class="fancy-card__strip" aria-hidden>
        <div
          v-for="[name, hex] in colors"
          :key="name"
          class="fancy-card__strip-tile"
          :style="{ background: hex }"
        />
      </div>
    </template>
    <slot />
    <template #footer>
      <slot name="footer" />
    </template>
  </BaseCard>
</template>

<script>
import BaseCard from '../BaseCard/BaseCard.vue';
import { extendedPalette } from '../../tailwind/theme';

export default {
  components: { BaseCard },
  props: {
    /** If true, adds additional decorative design elements to the card. So fancy! */
    ornate: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    colors() {
      return Object.entries(extendedPalette);
    },
  },
};
</script>

<style scoped>
.fancy-card >>> .base-card__content {
  @apply py-10;
}

.fancy-card--ornate >>> .base-card__content {
  @apply bg-no-repeat;
  background-image: url('https://images.ctfassets.net/tb3mx4vb7wqk/4fsKBW1o705lMy6eDbWTwp/c020d0d481dd33271be53f63be85588e/curly.png?h=250'),
    url('https://images.ctfassets.net/tb3mx4vb7wqk/1x5sp7BAGypPVFidnBNe8p/9d09b40ca32343ced4dbc18413bea354/spots.png?h=250');
  background-position: left top, right bottom;
}

.fancy-card__strip {
  @apply flex;
}

.fancy-card__strip-tile {
  @apply flex-grow h-3;
}
</style>
