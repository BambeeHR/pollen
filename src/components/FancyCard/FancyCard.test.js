import { mount } from '@vue/test-utils';
import FancyCard from './FancyCard.vue';

describe('FancyCard', () => {
  test('renders correctly', () => {
    const wrapper = mount(FancyCard, {
      slots: {
        default: 'Hello world',
        footer: 'This is the footer',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
