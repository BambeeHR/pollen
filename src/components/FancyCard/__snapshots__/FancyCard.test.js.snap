// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyCard renders correctly 1`] = `
<article
  class="base-card fancy-card"
>
  <header
    class="base-card__header"
  >
    <div
      aria-hidden=""
      class="fancy-card__strip"
    >
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(205, 224, 223);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(247, 242, 220);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(255, 170, 0);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(226, 89, 89);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(191, 66, 40);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(104, 0, 33);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(126, 36, 92);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(72, 29, 114);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(0, 40, 85);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(0, 104, 128);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(25, 145, 169);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(22, 167, 167);"
      />
      <div
        class="fancy-card__strip-tile"
        style="background: rgb(0, 72, 81);"
      />
    </div>
  </header>
   
  <section
    class="base-card__content"
  >
     Hello world
  </section>
   
  <footer
    class="base-card__footer"
  >
    This is the footer
  </footer>
</article>
`;
